### PR TITLE
prgenv-nvidia for todi

### PR DESCRIPTION
**Pull request by bcumming**
Add a basic nvhpc sdk prgenv-nvidia

TODO
- [ ] documentation
- [ ] test on todi

### CODE REVIEW

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=prgenv-nvidia:24.6

**Review by toxa81 — APPROVED** [review]
LGTM!

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=prgenv-nvidia:24.6

File: recipes/prgenv-nvidia/24.6/gh200/environments.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - cuda@12.4
+  - fftw
+  - fmt
+  - hdf5%nvhpc +fortran
+  - libtree
+  - meson
+  - nccl
+  - nccl-tests
+  - ninja
+  - openblas threads=openmp
```
Comment (simonpintarelli) [inline recipes/prgenv-nvidia/24.6/gh200/environments.yaml:23]: openblas is linked to `libgomp.so` isn't this creating a conflict with `nvomp.so`? Ie two different openmp providers being used?  Same for fftw afaik.

**Comment by bcumming** [comment]
cscs-ci run alps;system=todi;uarch=gh200;uenv=prgenv-nvidia:24.6

**Comment by msimberg** [comment]
Question that doesn't have to be resolved here, but is related: Is there a plan to include nvpl in this uenv or in other uenvs? From testing DLA-Future, nvpl performs as well or better than openblas. The downside is that not all packages support using nvpl for BLAS/LAPACK.

**Comment by bcumming** [comment]
We were discussing the choice between NVPL and OpenBLAS.
Frankly, it is a bit of a mess trying to pick a single implementation to provide in a uenv, because not only is NVPL not supported directly by all packages, but you probably have to provide OpenBLAS with and without OpenMP threading.

**Pull request closed without merging**